api/client.py

```diff
@@ -57,30 +61,87 @@ def generate(model_name, prompt, system=None, template=None, format="", context=
     except requests.exceptions.RequestException as e:
         print(f"An error occurred: {e}")
         return None, None
+    
+
+# Create a blob file on the server if it doesn't exist.
+def create_blob(digest, file_path):
+    url = f"{BASE_URL}/api/blobs/{digest}"
+
+    # Check if the blob exists
+    response = requests.head(url)
+    if response.status_code != 404:
+        return  # Blob already exists, no need to upload
+
+    # Upload the blob
+    with open(file_path, 'rb') as file_data:
+        requests.post(url, data=file_data)
+
 
 # Create a model from a Modelfile. Use the callback function to override the default handler.
-def create(model_name, model_path, callback=None):
+def create(model_name, filename, callback=None):
     try:
+        file_path = Path(filename).expanduser().resolve()
+        # Read and process the modelfile
+        with open(file_path, 'r') as f:
+            lines = f.readlines()
```

Comment:
Instead of reading the entire file into memory, you can use `f` as an iterator directly

```python
for line in f:
  line = line.strip()
  if not line:
    continue

  ...
```